### PR TITLE
ref(pkg/chartutil): decrease map lookups

### LINKED ISSUE #2663: Several map lookups in chart value coalescing procedure can be eliminated
In #2648, @scottrigby augmented the chart value coalescing procedure to treat null values specially, indicating that the given mapping should be removed. Those changes revealed [a few opportunities for removing map lookup expressions](https://github.com/kubernetes/helm/pull/2648#discussion_r126238947) in the [`chartutil.coalesceValues`](https://github.com/scottrigby/helm/blob/8be9bb51de31b551ab4178d6e43616fbbbca9850/pkg/chartutil/values.go#L263-L304) function—assuming that the Go compiler won't eliminate them already via CSE.

While we're there, consider [moving part of a comment referring to the YAML specification](https://github.com/kubernetes/helm/pull/2648#discussion_r126239691) to the unit test that makes use of the referenced definitions.

### PR DESCRIPTION
**Pull request by rbwsam**
Decrease map lookups in `chartutil.coalesceValues` and move comment to unit
test.

Closes #2663

### CODE REVIEW

File: pkg/chartutil/values_test.go [hunk 1]
```diff
@@ -350,6 +350,10 @@ func TestCoalesceValues(t *testing.T) {
 		}
 	}
 
+	// When the YAML value is null, we remove the value's key.
```
Comment (seh) [inline pkg/chartutil/values_test.go]: - [ ] I apologize for not being more clear in [my original suggestion](https://github.com/kubernetes/helm/pull/2648#discussion_r126239691), but what I meant was that we should move only line 356 (starting with "ref:") over here to clarify why we look at these different ways of writing null in YAML, and leave the rest of the comment in place in the `coalesceValues` function.  

  I'd put this specification reference at [line 277](https://github.com/rbwsam/helm/blob/51b59497eed3537340c80750680e6626145520fc/pkg/chartutil/values_test.go#L277), just above the `testCoalesceValuesYaml` variable.

**Review by seh — COMMENTED** [review]
Take my suggestion here as just that, not one to block a merge.

**Review by seh — APPROVED** [review]
Thank you!

**Comment by technosophos** [comment]
Feel free to merge this @seh

**Review by technosophos — APPROVED** [review]
LGTM2

**Pull request merged**